### PR TITLE
Refactor hook

### PR DESCRIPTION
**Pull request by scanon**
This refactor the hook configuration and integrates it with podman_hpc

Feature include:
* There is now a config directory that act like modules
* The configuration can specify command-line flags and help to add to podman_hpc (so options can be added without code changes)
* The configuration includes command-line options to add to podman that can't be handled in the hook.

### CODE REVIEW

File: etc/modules.d/mpich.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: mpich
+cli_arg: mpi
+help: Enable Cray MPICH Support
+env: ENABLE_MPICH_SS
+additional_args:
+  - -e SLURM_*
+  - -e PALS_*
+  - -e PMI_*
+  - -e LD_LIBRARY_PATH=/opt/udiImage/modules/mpich:/opt/udiImage/modules/mpich/dep
```
Comment (lastephey) [inline etc/modules.d/mpich.yaml:9]: just thinking about technical debt- should we make this a variable? when we retire shifter, maybe we'll want to use a different path?

File: podman_hpc/podman_hpc.py [hunk 2]
```diff
@@ -195,21 +146,35 @@ def get_params(args):
...
     parser.add_argument("--additional-stores", type=str,
                         help="Specify other storage locations")
     parser.add_argument("--squash-dir", type=str,
                         help="Specify alternate squash directory location")
     parser.add_argument("--update-conf", action="store_true",
                         help="Force update of storage conf")
+    confs = read_confs()
+    add_args(parser, confs)
     args, podman_args = parser.parse_known_args()
+    if "--help" in podman_args:
```
Comment (lastephey) [inline podman_hpc/podman_hpc.py:176]: +1 for adding help

File: podman_hpc/podman_hpc.py [hunk 3]
```diff
@@ -195,21 +146,35 @@ def get_params(args):
...
-                        help="Enable gpu support")
-    parser.add_argument("--mpich", action="store_true",
-                        help="Enable mpich support")
-    parser.add_argument("--hpc", action="store_true",
-                        help="Enable hpc support")
     parser.add_argument("--additional-stores", type=str,
                         help="Specify other storage locations")
     parser.add_argument("--squash-dir", type=str,
                         help="Specify alternate squash directory location")
     parser.add_argument("--update-conf", action="store_true",
```
Comment (lastephey) [inline podman_hpc/podman_hpc.py:171]: could we add any kind of "verbose" or "debug" option so the podman run statement gets printed somewhere?

**Comment by lastephey** [comment]
Thanks @scanon, I think it looks good. Just a few minor questions and comments.

File: etc/modules.d/mpich.yaml (hunk 1 above)
Comment (scanon) [reply etc/modules.d/mpich.yaml:9]: Yes.  I think we may be able to directly use the cray paths with podman.  We need to test that out.

File: podman_hpc/podman_hpc.py (hunk 3 above)
Comment (scanon) [reply podman_hpc/podman_hpc.py:171]: Good idea.  I'll make an issue for that.

**Pull request merged**